fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -372,7 +372,6 @@ def state_dict(self, *args, **kwargs):  # type: ignore
         torch.cuda.synchronize()
         self._lazy_init()
         self._rebuild_full_params()
-        self._all_buffers_to(dtype=torch.float32)  # Buffers dtype stays consistent with parameters.
```

Comment:
don't we still need this for FP typed buffer? It seems that we need to make all buffer fp32, then get state_dict from underlying modules and then put the buffers back in fp16 (if needed) on line 382, right?